model/project.go

```diff
@@ -1631,6 +1633,115 @@ func (p *Project) GetDisplayTask(variant, name string) *patch.DisplayTask {
 	return nil
 }
 
+func GetVersionsWithOptions(projectId string, opts GetVersionsOptions) ([]Version, error) {
+	match := bson.M{
+		VersionIdentifierKey: projectId,
+	}
+	if opts.ByBuildVariant != "" {
+		match[bsonutil.GetDottedKeyName(VersionBuildVariantsKey, VersionBuildStatusVariantKey)] = opts.ByBuildVariant
+	}
+	if opts.VersionToStartAt != "" {
+		v, err := VersionFindOneId(opts.VersionToStartAt)
+		if err != nil {
+			return nil, errors.Wrapf(err, "problem finding version '%s'", opts.VersionToStartAt)
+		}
+		if v == nil {
+			return nil, errors.Errorf("version '%s' doesn't exist", opts.VersionToStartAt)
+		}
+		match[VersionCreateTimeKey] = bson.M{"$lte": v.CreateTime}
+	}
+
+	pipeline := []bson.M{bson.M{"$match": match}}
+	pipeline = append(pipeline, bson.M{"$sort": bson.M{VersionCreateTimeKey: -1}})
+	project := bson.M{
```

Comment:
this project doesn't get used unless you're including builds. Is this intentional?